.ci/scripts/util.sh

```diff
@@ -2,65 +2,129 @@
 
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 
-metrics_collection_start ()
+# Start the metrics collection of both docker stats and the miscellaneous os statistics.
+# Parameters:
+#     container_info:  the name of the docker file from which to gather statistics
+#     sample_time:  The time between sampling of the statistics.
+metrics_collection_start()
 {
-    # container name should always contain pge_docker_tag, PGE, and runconfig (at a minimum)
-    container_name=$1
+    local pge=$1
+    # Split seconds argument into docker name and tag
+    local cont_name="$(echo "$2" | cut -d':' -f1)"
+    local container_tag="$(echo "$2" | cut -d':' -f2)"
+    # cont_info should always contain pge_docker_tag, PGE, and runconfig (at a minimum)
+    # container_info="${pge}_${cont_name}_${container_tag}"
 
+    local container_info="$pge"
+
+    # If no sample_time value is passed - default to a value of 1
+    if [[ -z "$2" ]]
+    then
+        local sample_time=1
+    else
+        local sample_time=$3
+    fi
+
+    echo "Using sample time of: $sample_time"
     # Initialize output files and statistics format
-    metrics_stats="${container_name}_metrics_stats.csv"
+    metrics_stats="${container_info}_metrics_stats.csv"
+    echo "local file name from assignment at line 30: $metrics_stats"
```

Comment:
Is this a leftover debug statement? Seems strange to have the hardcoded reference to line 30 here....

Reply:
Yes, however, this will all go away when I store intermediate data in memory rather than writing temp files.

Reply:
I would like to continue to see these files kept for a couple of reasons.  One reason is that it is the original record of the metrics which can be imported into excel for quick analysis, or can be re-processed by a script at a later date to create other types of reports.  An example of post-processing might be to create a graph showing a parallel processing PGE's run-time versus vCPU quantity; you might want to reference the raw metrics data from many docker container executions (in Excel or a script).
Another reason is if there is a program interruption, a file will capture at least some of the data.  I had this happen with a multi-hour job where metrics were collected, but the terminal connection to the AWS instance was lost before the process_metrics_data.py script was run.  I was able to get the information I needed from the partial file.

Reply:
I no longer delete those two files